File: pluck/routes/session.py
"""Session / wallet / pricing / capabilities."""
from fastapi import APIRouter, Request

from ..capabilities import capabilities
from ..mythos import consumer, wallet_balance, wallet_topup
from ..pricing import PRICING

router = APIRouter()


@router.get("/api/session")
async def api_session(request: Request):
    m = consumer(request)
    return {"user": m["displayName"], "email": m.get("email"),
            "balance": await wallet_balance(m["userId"]), "cost": PRICING["base"]}


@router.post("/api/topup")
async def api_topup(request: Request):
    m = consumer(request)
    return {"balance": await wallet_topup(m["userId"], 10)}


@router.get("/api/pricing")
def api_pricing():
    """Single source of truth for the client's live cost estimate."""
    return {"pricing": PRICING}


@router.get("/api/capabilities")
def api_capabilities():
    """Which optional features are installed (UI hides the rest)."""
    return capabilities()

File: node/src/routes/session.js
/** Session / wallet / pricing / capabilities. Mirrors pluck/routes/session.py. */
import { Router } from "express";

import { capabilities } from "../capabilities.js";
import { PRICING } from "../pricing.js";
import { wrap } from "./helpers.js";

export function createRouter(deps) {
  const { consumer, walletBalance, walletTopup } = deps;
  const router = Router();

  router.get("/api/session", wrap(async (req, res) => {
    const m = consumer(req);
    res.json({
      user: m.displayName,
      email: m.email ?? null,
      balance: await walletBalance(m.userId),
      cost: PRICING.base,
    });
  }));

  router.post("/api/topup", wrap(async (req, res) => {
    const m = consumer(req);
    res.json({ balance: await walletTopup(m.userId, 10) });
  }));

  /** Single source of truth for the client's live cost estimate. */
  router.get("/api/pricing", (_req, res) => {
    res.json({ pricing: PRICING });
  });

  /** Which optional features are installed (UI hides the rest). */
  router.get("/api/capabilities", (_req, res) => {
    res.json(capabilities());
  });

  return router;
}
